static/js/hash_util.js

```diff
@@ -23,8 +23,17 @@ export function get_hash_section(hash) {
 // Some browsers zealously URI-decode the contents of
 // window.location.hash.  So we hide our URI-encoding
 // by replacing % with . (like MediaWiki).
+// Also we encode parenthesis because they create issue when combined with markdown link format.
+// For Example:[test](https:\\testHost/test)/something),this link will break after "test".
 export function encodeHashComponent(str) {
-    return encodeURIComponent(str).replace(/\./g, "%2E").replace(/%/g, ".");
+    const characterToBeReplaced = {
+        "%": ".",
+        "(": ".28",
+        ")": ".29",
+    };
+    return encodeURIComponent(str)
+        .replace(/\./g, "%2E")
```

Comment:
Why doesn't this belong in the `characterToBeReplaced` map?

Reply:
`.` must be replaced first because it will create `%`("%2E"), and then these `%` also need to be replaced with`.`. 
For example, take the URL as https://www.zulipexample.com, now if we include `.` in characterToBeReplaced and then URL after encoding is "https.3A.2F.2Fwww%2Ezulipexample%2Ecom" but the correct encoding is "https.3A.2F.2Fwww.2Ezulipexample.2Ecom"

Reply:
The end result of that is that `.` gets turned into `.2E`, right? So wouldn't this produce the same result as adding `".": ".2E",` to `characterToBeReplaced`? (And adding `.` to the character class being matched in that `.replace` call.)

Reply:
Oh, this also works. I will update the PR. Thanks for the suggestion.

Reply:
@gnprice It seems like PR is closed. Can you please open it again?

Reply:
Thanks! Go ahead and open a new PR for this tweak (and just link to this comment from there.) Once a PR has already been merged, GitHub doesn't allow re-opening it, because the branch that the PR thread was about has already been merged.